shop/src/main/java/ro/msg/learning/shop/controller/RevenueController.java

```diff
@@ -0,0 +1,31 @@
+package ro.msg.learning.shop.controller;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.learning.shop.domain.entity.DailyRevenue;
+import ro.msg.learning.shop.service.revenue.RevenueService;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/api/v1/revenue")
+public class RevenueController {
+
+    private final RevenueService revenueService;
+
+    @GetMapping("/daily")
+    public ResponseEntity<List<DailyRevenue>> getRevenueForToday(@RequestParam String dateFormat, @RequestParam String date) throws ParseException {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat);
+        Date formatedDate = simpleDateFormat.parse(date);
```

Comment:
What happens here if the "date" is not in a correct format or this endpoint is called with a random string instead of a date? You need to at least throw a nice exception here if you don't want to change the type of the "date" parameter from String to Date.

Also, the name of the method is not reflecting the implementation, you get the revenue for a specific date (passed as parameter) not for today. It can be today but it can also be any date.